src/nxt_sockaddr.c

```diff
@@ -612,12 +610,19 @@ nxt_sockaddr_unix_parse(nxt_mp_t *mp, nxt_str_t *addr)
      *   are covered by the specified length of the address structure.
      *   (Null bytes in the name have no special significance.)
      */
-    if (path[0] == '@') {
+    switch (path[0]) {
+    case '@':
         path[0] = '\0';
+        /* fall through */
+    case '\0':
         socklen--;
-    }
-
+#if !(NXT_LINUX)
+        nxt_thread_log_error(NXT_LOG_ERR,
+                          "abstract unix domain socket \"%V\" is not supported",
+                          addr);
+        return NULL;
 #endif
+    }
```

Comment:
OK, so this enables improved error messages in !Linux, e.g testing on FreeBSD, using either '@magic' or '\u0000magic' for the socket name would result in

```JSON
{
        "error": "Failed to apply new configuration."
}
```
Now we get the likes of
```JSON
{
        "error": "Invalid configuration.",
        "detail": "The listener address \"unix:@magic\" is invalid."
}
```

So LGTM.

Reply:
Hmmm, did FreeBSD result in an error before this addition?  I was worried that without this, FreeBSD might try to create a file with an embedded `\0`.  If it's not the case, I prefer not having this extra code.  Especially since using abstract sockets in FreeBSD is nonsense, so I don't want to add much code to make it nicer.  I just want to make sure it fails hard, instead of doing weird stuff silently

Reply:
Could/did you try?

Reply:
OK, scrub that. If I revert that error check

- It will create a socket called '@magic' in the processes current working directory (if it can).
- It does error out with "Failed to apply new configuration." with '\u0000magic'.